UFDR parsing failed caused by too large contact name

Prezados,

Bom dia, meu nome é Luís Renato, sou do MP de São Paulo.
Realizei recentemente a extração de um telefone celular marca Samsung SM-A546E, a extração foi realizada pelo Cellebrite Inseytes versão 10.9.0.284, fiz o processamento no Physical Analyzer, versão 7.74.0.12 e o mesmo não montou as conversas de WhastApp. Ao fazer a indexação no IPED 4.3.1, o software identificou as conversas de WhatsApp e as montou perfeitamente, a indexação foi um sucesso.
Porém, ao falar com o Analista do caso, o mesmo informou que sempre utiliza o Celebrite Reader para analisar os casos e eu informei a ele que as conversas de WhatsApp só haviam sido identificadas pelo IPED.
Como ele informou dessa necessidade, resolvi processar novamente a aquisição de dados, só que desta vez utilizando o PA mais atual,  versão 10.9.0.3029, que montou as conversas de WhatsApp. Até aí, tudo bem, o problemas vieram quando resolvi fazer a indexação no IPED e a mesma acabou apresentando erro, que pode ser visto no log que anexei.
Quando vi o log pela primeira vez, vi que ele pedia para que eu fizesse a inclusão do Plugin JPEG2000. Baixei o arquivo "jai-imageio-jpeg2000-1.3.0.jar" em (https://mvnrepository.com/artifact/com.github.jai-imageio/jai-imageio-jpeg2000/1.3.0), conforme indicado pelo log. Inseri na pasta plugin e dei continuidade ao processo de indexação, que por sua vez apresentou erro novamente, ainda mencionando que o referido plugin não havia sido localizado.
Não consegui realizar a indexação.
Agradeço por sua atenção e gostaria de encontrar uma solução para o erro.

Att.

Luís Renato Correia Guimarães
luisguimaraes@mpsp.mp.br



[IPED-SearchApp.log](https://github.com/user-attachments/files/28712670/IPED-SearchApp.log)

Here is the relevant stack trace from the log provided by @luisguimaraes-MP:

```text
java.lang.Exception: Worker-3 Error while processing SEIXXX.ufdr/_DecodedData/Contact/Contact_54d7db30-3907-4ee6-a730-a28b4bbc61c9 (null bytes)
	at iped.engine.core.Worker.process(Worker.java:188)
	at iped.engine.core.Worker.run(Worker.java:267)
Caused by: java.lang.IllegalArgumentException: DocValuesField "name" is too large, must be <= 32766
	at org.apache.lucene.index.SortedDocValuesWriter.addValue(SortedDocValuesWriter.java:77)
	at org.apache.lucene.index.IndexingChain.indexDocValue(IndexingChain.java:977)
	at org.apache.lucene.index.IndexingChain.processField(IndexingChain.java:758)
	at org.apache.lucene.index.IndexingChain.processDocument(IndexingChain.java:620)
	at org.apache.lucene.index.DocumentsWriterPerThread.updateDocuments(DocumentsWriterPerThread.java:241)
	at org.apache.lucene.index.DocumentsWriter.updateDocuments(DocumentsWriter.java:432)
	at org.apache.lucene.index.IndexWriter.updateDocuments(IndexWriter.java:1532)
	at org.apache.lucene.index.IndexWriter.addDocuments(IndexWriter.java:1503)
	at iped.engine.task.index.IndexTask.process(IndexTask.java:149)
	at iped.engine.task.AbstractTask.processMonitorTimeout(AbstractTask.java:282)
	at iped.engine.task.AbstractTask.processAndSendToNextTask(AbstractTask.java:197)
	at iped.engine.task.AbstractTask.sendToNextTask(AbstractTask.java:230)
```

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/BaseModelHandler.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/BaseModelHandler.java
@@ -35,6 +35,8 @@ public class BaseModelHandler<T extends BaseModel> {
 
     private static final Logger logger = LoggerFactory.getLogger(BaseModelHandler.class);
 
+    protected static final int maxTitleLen = 4096;
+
     protected final T model;
     protected final IItemReader item;
 

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/ContactHandler.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/ContactHandler.java
@@ -95,7 +95,7 @@ public class ContactHandler extends AccountableHandler<Contact> {
                 .append(model.getModelType())
                 .append(typeStr)
                 .append("-[")
-                .append(StringUtils.firstNonBlank(data, model.getId()))
+                .append(StringUtils.truncate(StringUtils.firstNonBlank(data, model.getId()), maxTitleLen))
                 .append("]")
                 .toString();
     }

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/EmailHandler.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/EmailHandler.java
@@ -103,15 +103,10 @@ public class EmailHandler extends BaseModelHandler<Email> {
 
     @Override
     public String getTitle() {
-        String title = StringUtils.firstNonBlank(model.getSubject(), model.getId());
-        int maxTitleLen = 4096;
-        if (title != null && title.length() > maxTitleLen) {
-            title = title.substring(0, maxTitleLen);
-        }
         return new StringBuilder()
                 .append("Email") //
                 .append("-[") //
-                .append(title) //
+                .append(StringUtils.truncate(StringUtils.firstNonBlank(model.getSubject(), model.getId()), maxTitleLen)) //
                 .append("]") //
                 .toString();
     }

--- a/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/UserAccountHandler.java
+++ b/iped-parsers/iped-parsers-impl/src/main/java/iped/parsers/ufed/handler/UserAccountHandler.java
@@ -38,7 +38,7 @@ public class UserAccountHandler extends AccountableHandler<UserAccount> {
                 .append(model.getModelType())
                 .append(sourceStr)
                 .append("-[")
-                .append(StringUtils.firstNonBlank(data, model.getId()))
+                .append(StringUtils.truncate(StringUtils.firstNonBlank(data, model.getId()), maxTitleLen))
                 .append("]")
                 .toString();
     }

